classes/ramen_shop/RamenShopManager.php

```diff
@@ -0,0 +1,441 @@
+            shop_description: self::RAMEN_SHOP_DESCRIPTIONS_BY_VILLAGE[$village->village_id],
+            dialogue: self::RAMEN_SHOP_DIALOGUE_OPTIONS_BY_VILLAGE[$village->village_id][array_rand(self::RAMEN_SHOP_DIALOGUE_OPTIONS_BY_VILLAGE[$village->village_id])],
+            shop_name: self::RAMEN_SHOP_NAMES_BY_VILLAGE[$village->village_id],
+        );
+    }
+
+    /**
+     * Load the basic ramen details for the given player
+     * @param System $system
+     * @param User $player
+     * @return BasicRamenDto[]
+     */
+    public static function loadBasicRamen(System $system, User $player): array {
+        $ramen_cost_multiplier = 1 - ($player->village->active_upgrade_effects[VillageUpgradeConfig::UPGRADE_EFFECT_RAMEN_COST] / 100);
+        $rankManager = new RankManager($system);
+        $rankManager->loadRanks();
+
+        $health[1] = $rankManager->healthForRankAndLevel(1, $rankManager->ranks[1]->max_level);
+        $health[2] = $rankManager->healthForRankAndLevel(2, $rankManager->ranks[2]->max_level);
+        $health[3] = $rankManager->healthForRankAndLevel(3, $rankManager->ranks[3]->max_level);
+        $health[4] = $rankManager->healthForRankAndLevel(4, $rankManager->ranks[4]->max_level);
+        // $health[5] = $rankManager->healthForRankAndLevel(5, $rankManager->ranks[5]->max_level);
+
+        $basic_ramen[self::BASIC_RAMEN_SMALL] = new BasicRamenDto(
+            key: self::BASIC_RAMEN_SMALL,
+            cost: !$player->location->equals($player->village_location) ? $player->rank_num * 5 * 5 : ($player->rank_num * 5 * $ramen_cost_multiplier),
+            health_amount: $health[$player->rank_num] * 0.1,
+            label: 'Shio S',
+            image: "images/ramen/ShioS.png",
+        );
+        $basic_ramen[self::BASIC_RAMEN_MEDIUM] = new BasicRamenDto(
+            key: self::BASIC_RAMEN_MEDIUM,
+            cost: !$player->location->equals($player->village_location) ? $player->rank_num * 25 * 5 : ($player->rank_num * 25 * $ramen_cost_multiplier),
+            health_amount: $health[$player->rank_num] * 0.5,
+            label: 'Shio M',
+            image: "images/ramen/ShioM.png",
+        );
+        $basic_ramen[self::BASIC_RAMEN_LARGE] = new BasicRamenDto(
+            key: self::BASIC_RAMEN_LARGE,
+            cost: !$player->location->equals($player->village_location) ? $player->rank_num * 50 * 5 : ($player->rank_num * 50 * $ramen_cost_multiplier),
+            health_amount: $health[$player->rank_num] * 1,
+            label: 'Shio L',
+            image: "images/ramen/ShioL.png",
+        );
+
+        if ($system->isDevEnvironment()) {
+            foreach ($basic_ramen as $ramen) {
+                $ramen->cost = 0;
+            }
+        }
+
+        return $basic_ramen;
+    }
+
+    /**
+     * Load the special ramen details for the given player
+     * @param User $player
+     * @return SpecialRamenDto[]
+     */
+    public static function loadSpecialRamen(User $player): array {
```

Comment:
```suggestion
    public static function getSpecialRamen(User $player): array {
```

I think `load` implies we're either storing into the class or pulling from the DB, maybe we can use `get` instead?